test/tests/query_params_test.js

```diff
@@ -3,8 +3,24 @@ import Router from "router";
 import { resolve, configure, reject, Promise } from "rsvp";
 
 var router, url, handlers, expectedUrl, actions;
+var scenarios = [
+  {
+    name: 'Sync Get Handler',
+    getHandler: function(name) {
+      return handlers[name] || (handlers[name] = {});
+    }
+  },
+  {
+    name: 'Async Get Handler',
+    getHandler: function(name) {
+      return Promise.resolve(handlers[name] || (handlers[name] = {}));
```

Comment:
We chatted about this in slack, and decided to leave these looped tests as is and add a single new async only test that uses `setTimeout` and confirms a transition into a lazily resolved route works properly.

Future work will be to refactor these tests to not assume that all async is pseudo/fake async in the BB queue.